### ISSUE TITLE
Using ";k" to insert "kappa", the last "a" is automatically converted to "k"

### ISSUE THREAD

**Issue by K-gihu** [tecosaur/LaTeX-auto-activating-snippets#20]
To insert `\kappa`, I inserted `;k` and saw the last letter `a` is converted to `k` automatically.

**Comment by ymarco** [comment]
You type ;k and get ;kappk?
I can't reproduce, some more info would be nice.
I suspect that another post-self-insert-hook is messing the buffer up after aas-post-self-insert-hook runs.

**Comment by K-gihu** [comment]
My `post-self-insert-hook` is `'(aas-post-self-insert-hook t)` and nothing else. 
I tried using laas-mode in vanilla emacs by `emacs -Q` and this problem disappeared. 
It should be related with my own configuration. I'll find out the reason later.

Here is a screencast.
![Peek 2021-07-06 20-37](https://user-images.githubusercontent.com/49723096/124603371-71c60780-de9c-11eb-9d03-cabcfaee97f8.gif)

I am using doom configuration and here the `doom/info` output:
```
generated    7月 06, 2021 20:42:35
system       Arch Linux Linux 5.12.14-arch1-1 x86_64 x
emacs        28.0.50 master 09081ca11 ~/doom-emacs/ -> ~/doom-emacs/
doom         2.0.9 grafted, HEAD -> develop, origin/develop, origin/HEAD, _upgrade_develop 27316850 2021-06-07 21:16:29 -0400 ~/.config/doom/ -> ~/.config/doom/
shell        /bin/zsh
features     ACL CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GPM GSETTINGS HARFBUZZ JPEG JSON LCMS2 LIBSYSTEMD LIBXML2 MODULES NATIVE_COMP NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND THREADS TIFF TOOLKIT_SCROLL_BARS X11 XDBE XIM XPM GTK3 ZLIB
traits       gui server-running chemacs
modules      :completion (company +childframe) ivy :ui doom doom-quit (emoji +unicode) hl-todo hydra modeline nav-flash ophints (popup +defaults +all) treemacs vc-gutter vi-tilde-fringe :editor (evil +everywhere) (format +onsave) multiple-cursors snippets word-wrap :emacs dired electric ibuffer (undo +tree) vc :checkers (syntax +childframe) :tools biblio debugger (eval +overlay) lookup lsp magit pdf :lang (cc +lsp) emacs-lisp latex (org +roam +journal +present +noter +dragndrop) plantuml sh yaml :config literate (default +bindings)
packages     (sis) (evil-pinyin) (evil-textobj-line) (org-noter-pdftools) (org-fragtog) (pangu-spacing) (org-roam-server) (command-log-mode) (auto-activating-snippets :recipe (:host github :repo ymarco/auto-activating-snippets)) (latex-auto-activating-snippets :recipe (:host github :repo tecosaur/LaTeX-auto-activating-snippets)) (org-super-agenda) (valign) (keyfreq) (lister :recipe (:host github :repo publicimageltd/lister)) (delve :recipe (:host github :repo publicimageltd/delve)) (color-rg :recipe (:host github :repo manateelazycat/color-rg)) (centaur-tabs) (auto-save :recipe (:host github :repo manateelazycat/auto-save)) (wucuo) (eaf :recipe (:host github :repo manateelazycat/emacs-application-framework :files (*))) (ctable :recipe (:host github :repo kiwanami/emacs-ctable)) (epc) (ox-rst) (xref-rst) (pyim)
```

My doom configuration file can be viewed [here](https://github.com/K-gihu/dotfiles/blob/c17d3a347c300ca6dfe641c5cbfa15dd2dde8160/.config/doom/config.el#L647-L688)

**Comment by ymarco** [comment]
Huh, the k appears with a delay after kappa is inserted.
Does it happen when you type something new, or just on it's own a few seconds after ;k is expanded?

If it's by some idle timer, it's probably caused by another package somehow. aas/laas don't do anything with idle timers.

I'll try to run your config in a few days and debug if things aren't solved by then.

**Comment by K-gihu** [comment]
OK, I think this is somehow related to `evil-escape-key-sequence` after testing.
My testing procedure is:
1. use `aas-set-snippet` to set new snippets of `;k` to `\kappa`, `kapp`, `kap` and `ka`. They all have their last letter converted to k.
2. Since I am using evil, I think it might be related to my `evil-escape-key-sequence` which was set to "jk". So I set some snippets of `;j` and same problem occured, just as I expected.
3. I set `evil-escape-key-sequence` to "fd" and this time `;f` showed the problem, while `;j` and `;k` went back to normal.

**Comment by K-gihu** [comment]
After further testing, I found the `evil-escape-unordered-key-sequence` variable is related to the problem.
In my configuration file, I have set that variable to `t` and have also set `evil-escape-delay` variable, two variables in all. I found commenting or uncommenting the setting of `evil-escape-delay` does not affect the problem, but commenting out `evil-escape-unordered-key-sequence` will eliminate the problem.
